app/javascript/types/index.ts

```diff
@@ -1,5 +1,9 @@
 export * from '@/types/serializers';
 
+export type Intersection<T, U> = {
+  [K in Extract<keyof T, keyof U>]: T[K];
+};
```

Comment:
I use this type heavily in this PR; it gives a type that has only the common properties of its two arguments. (TypeScript has the `&` "intersection" operator, but I would call it a "merge" rather than an "intersection"; it combines the properties from both types, whether or not they are shared in common.)

Reply:
The format that I conventionally used is `Intersection<ModelTypeFromTypelizer, ResponseTypeFromJsonSchema>`.